cdap-unit-test/src/test/java/io/cdap/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -1036,6 +1036,57 @@ public void testGetServiceURL() throws Exception {
     centralServiceManager.waitForStopped(10, TimeUnit.SECONDS);
   }
 
+  @Category(SlowTests.class)
+  @Test
+  public void testGetServiceURLDiffNamespace() throws Exception {
+    ApplicationManager discoveringApplicationManager = deployApplication(AppUsingGetServiceURL.class);
+    ApplicationManager systemApplicationManager = deployApplication(NamespaceId.SYSTEM, AppUsingGetServiceURL.class);
+
+    // Discover system service from the default namespace
+    ServiceManager discoveringServiceManager = discoveringApplicationManager
+      .getServiceManager(AppUsingGetServiceURL.FORWARDING).start();
+    discoveringServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+    ServiceManager systemCentralServiceManager = systemApplicationManager
+      .getServiceManager(AppUsingGetServiceURL.CENTRAL_SERVICE).start();
+    systemCentralServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+    String result = callServiceGet(discoveringServiceManager.getServiceURL(),
+                                   "forward/" + NamespaceId.SYSTEM.getNamespace());
+    result = new Gson().fromJson(result, String.class);
+    Assert.assertEquals(AppUsingGetServiceURL.ANSWER, result);
+
+    discoveringServiceManager.stop();
+    systemCentralServiceManager.stop();
+
+    discoveringServiceManager.waitForStopped(10, TimeUnit.SECONDS);
+    systemCentralServiceManager.waitForStopped(10, TimeUnit.SECONDS);
+
+    // Attempt to discover system service from the default namespace before the system service was started
+    discoveringServiceManager = discoveringApplicationManager.getServiceManager(AppUsingGetServiceURL.FORWARDING)
+      .start();
```

Comment:
you don't need to start and stop multiple times in this test. Just make the relevant calls earlier in the test.